docs/modules/devicetwin.md

```diff
@@ -0,0 +1,328 @@
+
+# DeviceTwin
+
+
+## Overview
+
+DeviceTwin is responsible for storing device status and syncing device status to the cloud. 
+It also provides query interfaces for applications.
```

Comment:
what kind of query interface ? can we have a link to a section below which describes these interfaces ?

Reply:
we can put a block diagram which depicts what are the submodules of device twin . each sub module's job can be expanded in later sections with more diagrams/images. we can do this in next PR incrementally.

Reply:
sure @rohitsardesai83 , In the subsequent PRs we will incrementally add diagrams, images or flowcharts in order to make it easier for developers to understand the device twin module.